### PR TITLE
[3217] Add support for message delivery indicator in the message list

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

Implement message delivery indicator in the message list.

### 🎨 UI Changes

https://user-images.githubusercontent.com/9600921/158708776-7ff004cc-4563-459d-8781-31c3064d2ec0.mp4

### 🧪 Testing

1. Send a message
2. Read the message from another client
3. Observe that the message was read

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/MessageReadStatusIcon.kt [hunk 1]
```diff
@@ -11,51 +11,76 @@ import io.getstream.chat.android.client.models.Message
...
+) {
+    val syncStatus = message.syncStatus
 
-        Icon(
-            modifier = modifier,
-            painter = painterResource(id = messageIcon),
-            contentDescription = null,
-            tint = iconTint,
-        )
+    when {
```
Comment (filbabic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/MessageReadStatusIcon.kt:59]: Idk if this is less code but maybe sth like:

```
val color = when {
  isMessageRead ->
  syncStatusCheck ->
  syncStatusCheck2 ->
}

val icon = when { 
...
}

Icon(
  modifier = modifier,
  icon = icon,
  tint = tint,
  contentDescription = null
)
```

Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/MessageReadStatusIcon.kt:59]: There are also other statuses like "failed permanently" that we display without delivery indicator at all. So, the snippet above will have an "if" check before the Icon.

Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/MessageReadStatusIcon.kt:59]: yeah that sounds good - if you feel like it's a bit more readable or sth we can go in that direction, if it's not, then I'm fine with the current impl.

**Review by filbabic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/MessageReadStatusIcon.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/MessageReadStatusIcon.kt:59]: 👍

**Pull request merged**